product/opni/components/dialogs/AddTokenDialog.vue

```diff
@@ -70,25 +126,74 @@ export default {
     name="add-token-dialog"
     styles="background-color: var(--nav-bg); border-radius: var(--border-radius); max-height: 100vh;"
     height="auto"
-    :width="400"
+    :width="600"
     :scrollable="true"
     @closed="close()"
   >
     <Card class="prompt-restore" :show-highlight-border="false" title="Create Token">
-      <h4 slot="title" class="text-default-text" v-html="'Create Token'" />
-      <div slot="body" class="pt-10">
-        <div class="row mb-10">
-          <div class="col span-12">
-            <LabeledInput v-model.trim="name" label="Name (Optional)" />
+      <h2 slot="title" class="text-default-text" v-html="'Create Token'" />
+      <div slot="body">
+        <div class="pt-10 pb-10">
+          <div class="row mb-10">
+            <div class="col span-12">
+              <LabeledInput v-model.trim="name" label="Name (optional)" />
+            </div>
+          </div>
+          <div class="row">
+            <div class="col span-12">
+              <LabeledSelect
+                v-model="expiration"
+                label="Expiration"
+                :options="expirationOptions"
+              />
+            </div>
+          </div>
+          <div v-if="expiration === 'custom'" class="row span-6 mt-5 expiry">
+            <input
+              v-model="customExpirationValue"
+              type="number"
+              :mode="'edit'"
+            >
+            <Select
+              v-model="customExpirationUnit"
+              class="ml-10"
+              :options="customExpirationUnitsOptions"
+              :clearable="false"
+            />
```

Comment:
We've typically only used unlabeled inputs if they occur in a table or list. I'll have to see how this looks later but it's something we may change.